src/v/datalake/record_multiplexer.h

```diff
@@ -55,20 +56,22 @@ class record_multiplexer {
     ss::future<result<write_result, writer_error>> end_of_stream();
 
 private:
+    ss::future<result<void, writer_error>>
+    mux_record(const model::record_batch_header&, model::record&&);
+
+    void advance_result_offset(kafka::offset offset);
+
     // Handles the given record components of a record that is invalid for the
     // target table.
     // TODO: this just drops the data. Consider a separate table entirely.
-    ss::future<result<std::nullopt_t, writer_error>> handle_invalid_record(
-      kafka::offset,
-      std::optional<iobuf>,
-      std::optional<iobuf>,
-      model::timestamp,
-      chunked_vector<std::pair<std::optional<iobuf>, std::optional<iobuf>>>);
+    ss::future<result<void, writer_error>>
+    handle_invalid_record(const model::record_batch_header&, model::record&&);
```

Comment:
This is probably a bad idea. I'll need to re-build header kvs in there as well...